src/backend/main.ts

```diff
@@ -2,64 +2,94 @@
+    : minimist(process.argv.slice(1));
+
+  const samplePath = app.isPackaged
+    ? path.join(app.getAppPath(), "build", "assets").replace("app.asar", "app.asar.unpacked")
+    : path.join("assets", "Baytown.bim");
 
-// Setup logging immediately to pick up any logging during IModelHost.startup()
-// Logger.initializeToConsole();
-// Logger.setLevelDefault(LogLevel.Warning);
-// Logger.setLevel(AppLoggerCategory.Backend, LogLevel.Info);
+  const iModel = getAppEnvVar("IMODEL");
+  const name = getAppEnvVar("PROJECT");
+
+  return {
+    sampleiModelPath: samplePath,
+    snapshotName: parsedArgs._[0] ?? getAppEnvVar("SNAPSHOT"),
+    clientId: getAppEnvVar("CLIENT_ID") ?? appInfo.id,
+    redirectUri: getAppEnvVar("REDIRECT_URI") ?? `http://localhost:3000/signin-callback`,
+    project: (name && iModel) ? { iModel, name } : undefined,
+  }
+}
 
 /**
  * Initializes Electron backend
  */
-async function initialize(rpcs: RpcInterfaceDefinition[]) {
-  let manager: StandardElectronManager;
-  if (process.env.NODE_ENV === "development")
-    manager = new WebpackDevServerElectronManager(3000); // port should match the port of the local dev server
-  else
-    manager = new IModelJsElectronManager(path.join(__dirname, "..", "..", "build"));
+const initialize = async () =>  {
+
+  // Setup logging immediately to pick up any logging during IModelHost.startup()
+  Logger.initializeToConsole();
+  Logger.setLevelDefault(LogLevel.Warning);
+  Logger.setLevel(AppLoggerCategory.Backend, LogLevel.Info);
+
+  const hostConfig = new IModelHostConfiguration();
+  hostConfig.applicationType = ApplicationType.NativeApp;
+  await IModelHost.startup(hostConfig);
+
+  // Initialize Presentation
+  Presentation.initialize();
+
+  // Initialize ElectronRpcManager with correct RPC interfaces
+  ElectronRpcManager.initializeImpl({}, getSupportedRpcs());
+
+  const opts: ElectronManagerOptions = {
+    webResourcesPath: path.join(__dirname, "..", "..", "build"),
+  };
+
+  const manager = (process.env.NODE_ENV === "development") ? new WebpackDevServerElectronManager(opts) : new IModelJsElectronManager(opts);
 
   await manager.initialize({
     width: 1280,
     height: 800,
     show: false,
   });
-  // tell ElectronRpcManager which RPC interfaces to handle
-  ElectronRpcManager.initializeImpl({}, rpcs);
+
   if (manager.mainWindow) {
     manager.mainWindow.show();
     if (process.env.NODE_ENV === "development")
       manager.mainWindow.webContents.toggleDevTools();
   }
-}
 
-// eslint-disable-next-line @typescript-eslint/no-floating-promises
-(async () => {
-  try {
-    // Initialize iModelHost
-    await IModelHost.startup();
-
-    // Initialize Presentation
-    Presentation.initialize();
-
-    // get RPCs supported by this backend
-    const rpcs = getSupportedRpcs();
-    await initialize(rpcs);
+  // register handlers for viewer's IPC methods.
+  ipcMain.handleOnce(appIpc("getConfig"), async () => getFrontendConfig()); // may only be called once

```

Comment:
- why is this async?
- why can it only be called once?

Reply:
> * why is this async?

Doesn't technically need to be but shouldn't hurt anything either.  The more we allow the event loop to free up the more responsive the frontend will be.

> * why can it only be called once?

We only call it once during initialization but I can switch it since you may want to look at the environment variables set in the backend again.

Reply:
But without any awaits, I don't think you're actually freeing up the loop. But okay.

My concern was more about the comment... this sounded like you were artificially limiting this. If that's not the case, then I'm not sure you need a comment at all?

Reply:
Good call on the comment.  I'll remove it.